dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -240,21 +599,52 @@ PrehandleResult KVStore::preHandleSSTsToDTFiles(
                 snaps,
                 proxy_helper,
                 tmt,
+                std::nullopt,
                 DM::SSTFilesToBlockInputStreamOpts(opt));
 
+            // `split_keys` do not begin with 'z'.
+            auto split_keys = getSplitKey(log, this, new_region, sst_stream);
+
             ReadFromStreamResult result;
-            std::tie(result, prehandle_result)
-                = executeTransform(new_region, prehandle_task, job_type, storage, sst_stream, opt, tmt);
+            if (split_keys.empty())
+            {
+                LOG_INFO(
+                    log,
+                    "Single threaded prehandling for single big region, range={}, region_id={}",
```

Comment:
```suggestion
                    "Single threaded prehandling for this region, range={}, region_id={}",
```

Reply:
There is another log says `Parallel prehandling for single big region ...`, the `single big` is just a specific string for identification